### ISSUE TITLE
Implement surrender level for battles in the battles module to support the NPC module changes.

### ISSUE THREAD

**Issue by nicholaswlee** [uchicago-cs/chiventure#1448]
The goal of this pull request is to implement surrender level into stat_t for the NPC Module to accommodate this [issue](https://github.com/uchicago-cs/chiventure/pull/1400). This would actually allow battles to end when the NPC's hp has gone below the surrender level. We would also like to return a new identifier for when a NPC surrenders as opposed to being defeated.

**Comment by nicholaswlee** [comment]
The goal of this PR is to allow NPC's to surrender when they reach a certain hp level. This will change two data types:
stat_t will now have surrender_level. 
```
/* stats stub */
typedef struct stat {
    int max_sp; //The maximum amount of skill/special points this unit can currently have.
    int sp; //The current amount of skill/special points of the player; magic and skills deplete this resource.
    int speed;
    int phys_atk; //The physical damage capacity of the unit; used in damage calculation against phys_def.
    int mag_atk; //The magical damage capacity of the unit; used in damage calculation against mag_def.
    int phys_def; //The physical damage resistance of the unit; calculated against phys_atk.
    int mag_def; //The magical damage resistance of the unit; calculated against mag_atk;
    int crit; //The default odds of this unit getting bonus damage; the value is out of 100.
    int accuracy; //The default odds of any move used by this unit hitting the target; the value is out of 100.
    int hp; //The current health of the unit; the battle ends when this hits 0.
    int max_hp; //The maximum amount of health this unit can currently have.
    int surrender_level; //The hp value that will cause a NPC to surrender. Default value is 0
    int xp; //The current amount of experience points gained by this unit.
    int level; //The current level of the unit.
} stat_t;
```
Additionally, we will also be changing battle_status_t.
```
/* gives the status of a battle */
typedef enum battle_status
{
    BATTLE_IN_PROGRESS,
    BATTLE_VICTOR_PLAYER,
    BATTLE_VICTOR_ENEMY,
    BATTLE_ENEMY_SURRENDER,
    NO_BATTLE
} battle_status_t;
```

The main two functions that need to change are:
- battle_flow_move() //This needs to incorporate surrender level
- print_battle_winner() //This needs to print a new statement for when an opponent surrenders

There may be other functions that might need to change, but this is a preliminary starting point.

**Comment by nicholaswlee** [comment]
Because this is a smaller pull request, we have decided to add tests to make the process a bit easier for when the battles test team creates their tests.

**Comment by nmkhan100** [comment]
We have modified the stat_t struct by introducing a new field, the surrender_level field, and we have changed the battle_status enum so it now includes BATTLE_ENEMY_SURRENDER. Because of these changes, we also changed two functions, the battle_flow_move() function which now includes the surrender level, and the print_battle_winner() function which now prints a statement for when an opponent surrenders. These are the the functions we have modified, though there may be more that we must change in the future. And to make testing easier, we have added new tests based on our changes.

Here is a link to the changes we're supporting: (https://github.com/uchicago-cs/chiventure/pull/1400)
